arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -1020,6 +1023,8 @@ void RocksDBEngine::beginShutdown() {
     _replicationManager->beginShutdown();
   }
 
+  // from now on, all started compactions will be canceled
+  ::cancelCompactions.store(true, std::memory_order_relaxed);
```

Comment:
Is it ok if other threads see some effects of `beginShutdown` early than they see `cancelCompactions` is `true`?
If not I think u should use `release`, overwise please add a comment.

Reply:
I think this `canceled` flag is just a hint that allows RocksDB to abort a running compaction somewhere in the middle. It is not really time-critical, because it is only a hint and is not even guaranteed to be honored.
But I will change the memory order here.

Reply:
I make mistake, I mean `later` not `early`. Question not about _speed_, but about the correctness

Reply:
Once we are in shutdown mode, it is fine for this flag to be picked up by anything in RocksDB. We basically want to cancel pending compactions eventually, so they don't block shutdown.

Reply:
Since RocksDB actually uses `memory_order_acquire` to read the value, it probably makes sense to use `memory_order_release` for the store, although I doubt that we need to establish any happens-before relation via this variable...